aspnet/mvc/views/layout.rst

```diff
@@ -0,0 +1,98 @@
+
+The default Visual Studio ASP.NET MVC project template defines a layout file named ``_Layout.cshtml`` in the ``Views/Shared`` folder:
+
+.. image:: layout/_static/web-project-views.png
+
+This layout defines a top level template for views in the app. Apps do not require a layout, and apps can define more than one layout, with different views specifying different layouts.
+
+An example ``_Layout.cshtml``:
+
+.. literalinclude:: ../../../common/samples/WebApplication1/src/WebApplication1/Views/Shared/_Layout.cshtml
+  :language: html
+  :emphasize-lines: 42,66
+
+Specifying a Layout
+-------------------
+
+Razor views have a ``Layout`` property. Individual views specify a layout by defining this property:
+
+.. literalinclude:: ../../../common/samples/WebApplication1/src/WebApplication1/Views/_ViewStart.cshtml
+  :language: html
+  :emphasize-lines: 2
+
+The layout specified can use a full path (example: ``/Views/Shared/_Layout.cshtml``) or a partial name (example: ``_Layout``). When a partial name is provided, the :doc:`razor view engine <razor>` will search for the layout file using its standard discovery process. That is, the controller-associated folder is searched first, followed by the ``Shared`` folder. This discovery process is identical to the one used to discover :doc:`partial views <partial>`.
+
+Layouts can be nested; each layout can itself specify another layout. When rendering a view, :ref:`ViewStart <viewstart>` is run first, followed by the view itself, followed by its layout(s).
+
+Every layout must call ``RenderBody`` . Wherever the call to ``RenderBody`` is placed, the contents of the view will be rendered.
+
+Sections
+^^^^^^^^
+
+A layout can optionally reference one or more *sections*, by calling ``RenderSection``. Each call to ``RenderSection`` can specify whether that section is required or optional. Individual views specify the content to be rendered within a section using the ``@section`` razor syntax. If a view defines a section, it must be rendered (or an error will occur).
+
+An example ``@section`` definition in a view:
+
+.. literalinclude:: ../../../common/samples/WebApplication1/src/WebApplication1/Views/Manage/VerifyPhoneNumber.cshtml
+  :language: html
+  :lines: 28-30
+
+Sections only flow from views. They cannot be referenced from partials, view components, or other parts of the view system.
+
+.. _viewimports:
+
+Importing Shared Directives
+---------------------------
+
+Views can use razor directives to specify namespaces or perform :doc:`dependency injection <dependency-injection>`. Directives used by many views may be specified in a ``_ViewImports.cshtml`` file. The ``_ViewImports`` file supports the following directives:
+
+- addTagHelper
```

Comment:
Agreed but I think we need to wait until such articles exist in order to do so effectively.